system/lib/pthread/library_pthread.c

```diff
@@ -177,20 +177,12 @@ void emscripten_async_waitable_close(em_queued_call* call) {
 
 extern double emscripten_receive_on_main_thread_js(int functionIndex, int numCallArgs, double* args);
 extern int _emscripten_notify_thread_queue(pthread_t targetThreadId, pthread_t mainThreadId);
-extern int __pthread_create_js(struct pthread *thread, const pthread_attr_t *attr, void *(*start_routine) (void *), void *arg);
 
 static void _do_call(void* arg) {
   em_queued_call* q = (em_queued_call*)arg;
   // C function pointer
   assert(EM_FUNC_SIG_NUM_FUNC_ARGUMENTS(q->functionEnum) <= EM_QUEUED_CALL_MAX_ARGS);
   switch (q->functionEnum) {
-    case EM_PROXIED_PTHREAD_CREATE:
-      q->returnValue.i =
-        __pthread_create_js(q->args[0].vp, q->args[1].vp, q->args[2].vp, q->args[3].vp);
-      break;
-    case EM_PROXIED_CREATE_CONTEXT:
-      q->returnValue.i = emscripten_webgl_create_context(q->args[0].cp, q->args[1].vp);
-      break;
     case EM_PROXIED_RESIZE_OFFSCREENCANVAS:
```

Comment:
It looks like the JSEvents dependency comes from this function.. did you happen to looking into removing this too?  Perhaps as a followup?

Reply:
Commit b3beb21d7d1bc0fdaf5eff6914a11501332952b1 stubs out this dependency if not necessary. If necessary, I can split this into a separate PR.

Reply:
Very cool.  I think this can probably land all together, but I'm always splitting things up if you have time.  For example it would be good to see the standalone effect of just introducing the `emscripten_set_offscreencanvas_size` function (with a stub version).

Reply:
OK, reverted for now with commit 74468caea8bd90760285bd0019d51c602e429214. I'll make another PR with this (after this one lands, to avoid conflicts in the code size tests).